### DISCUSSION TITLE
Server timed out

### DISCUSSION THREAD

**Discussion by Traphouse3** [fhdm-dev/scoppy#25] [Q&A] [answered]
Having issues pulling the .uf2 to install on my pico. Direct click in browser on Windows 10 times out, on Kali git clone times out, wget fails, and apt-get cannot locate package. Is there somewhere to grab the source code to compile in Code-OSS? Or another DL location?

**Comment by fhdm-dev** [comment, marked as answer]
Sorry about that. My server was down. Could you try again. Thanks.

**Reply by Traphouse3** [reply]
I got it! Thanks for the quick reply. I am kind of new to building things such as this but want to build a tiny pocket scope for automotive diag that I can keep on my keychain. Most sensors in automotive applications are 5v but some pull up to 12v. I am having a hard time grasping how to scale a variable voltage down to 3.3v but also keep it accurate. I am capable of building the frontend I just dont quite know how to go about it. Any ideas?

**Reply by fhdm-dev** [reply]
There are some [example frontend designs](https://github.com/fhdm-dev/scoppy/wiki/Analog-Front-End-Examples) on the Wiki. I think something like design 1 would work for you. If you're only measuring positive voltages then you could leave out the op-amp that provides the offset voltage (vref) and just connect Rin2 to ground. You'll also need to tweak the resistor values so that 12V gets attenuated to 3.3V.

If you then find you don't have enough accuracy for the 5V sensors you'll wan't to be able to switch the value of Rin1 (or Rin2) so that your 5V input is attenuated to 3.3V. You'll also need to 'tell' the app which voltage range is selected so that it shows the correct voltage level in the app. The method for doing this is described [here](https://github.com/fhdm-dev/scoppy/wiki/Analog-Front-End). I'm planning on publishing a front end design that does just this sort of thing.

Design 3 shows how to add some under/over-voltage protection. That might be worth looking at. I assume you're not planning on using this where there is a risk of high voltage pulses from ignition coils or the like. I wouldn't want your pico to get damaged - or even worse - your phone.

I hope that helps.
